lib/puppetserver/ca/utils/file_system.rb

```diff
@@ -50,6 +50,17 @@ def self.check_for_existing_files(one_or_more_paths)
           errors
         end
 
+        def self.symlink_to_old_cadir(current_cadir)
+          old_cadir = '/etc/puppetlabs/puppet/ssl/ca'
+          new_cadir = '/etc/puppetlabs/puppetserver/ca'
+          return if current_cadir != new_cadir
+          # Files should have already been moved if we're using the new cadir
+          # so it should be safe to forcibly remove it (which we need to do
+          # in order to create a symlink).
+          FileUtils.remove_dir(old_cadir, true)
+          FileUtils.symlink(new_cadir, old_cadir)
```

Comment:
These two calls can be combined into a single one using the [force](https://ruby-doc.org/stdlib-2.5.0/libdoc/fileutils/rdoc/FileUtils.html#method-i-ln_s) option.

```suggestion
FileUtils.symlink(new_cadir, old_cadir, force: true)
```

Reply:
I don't think that will do what we want. If the target is a directory, the symlink will be nested under it. So we'd end up with `$old_cadir/$new_cadir -> $new_cadir`. I just tried this out in irb and it seems consistent with [the docs](https://ruby-doc.org/stdlib-2.7.1/libdoc/fileutils/rdoc/FileUtils.html#method-c-ln_s), but let me know if I've missed something.

Reply:
Oh blerg, I misread that doc, good thing you tested it!